ark/connection.py

```diff
@@ -0,0 +1,111 @@
+import logging
+import os.path
+
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+class Connection(object):
+    _nethash = None
+    _version = None
+
+    def __init__(self, client, host, port=None, nethash=None, version=None):
+        self.session = requests.Session()
+
+        self.host = host
+        self.port = port
+        self.nethash = nethash
+        self.version = version
+        self.client = client
+
+        self.session.headers.update({
+            'port': '1',
+            'API-Version': self.client.api_version.replace('v', '')
+        })
+
+    @property
+    def nethash(self):
+        return self._nethash
+
+    @nethash.setter
+    def nethash(self, nethash):
+        self._nethash = nethash
+        if nethash:
+            self.session.headers['nethash'] = nethash
+        else:
+            try:
+                del self.session.headers['nethash']
+            except KeyError:
+                pass
+
+    @property
+    def version(self):
+        return self._version
+
+    @version.setter
+    def version(self, version):
+        self._version = version
+        if version:
+            self.session.headers['version'] = version
+        else:
+            try:
+                del self.session.headers['version']
```

Comment:
why do we delete the version?

Reply:
In case you set `version` to `None`, we delete it from headers as I don't see a point in passing `version` in headers if it's none. Also it might cause problems if version is set to none and API expects it to be set to something or not be there at all. Same goes for nethash